keras_cv/layers/preprocessing/invert.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+
+class Invert(tf.keras.layers.Layer):
+    """Applies (max_value - pixel + min_value) for each pixel in the image.
+
+    Args:
+        min_value: int or float. Lower bound of input pixel values.
+        max_value: int or float. Upper bound of input pixel values.
+
+    Usage:
+
+    ```python
+        (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+        print(images[0, 0, 0])  # [59 62 63]
+        # Note that images are Tensor with values in the range [0, 255]
+        invert = Invert(min_value=0, max_value=255)
+        images = invert(images)
+        print(images[0, 0, 0])  # [196, 193, 192]
+    ```
+
+    Call arguments:
+        images: Tensor of type int or float, with pixels in
+            range [`min_value`, `max_value`] and shape [batch, height, width, channels]
+            or [height, width, channels].
+    """
+
+    def __init__(self, min_value=0, max_value=255):
```

Comment:
IMO min_value and max_value should be required parameters - this will prevent people from passing floats, oblivious to the internal behavior of the layer.

Reply:
opinon @qlzh727 @fchollet ?

Reply:
Sorry - I mean't passing differently ranged values.

Reply:
Perfect, thanks!

Reply:
The layer would then start by clipping to the value range, then apply `max_value - x` ?

Reply:
if you're in the range [-1, 1] a solarize operation is actually:

```
(2 - (v + 1)) - 1
```

Right?  If I understand correctly you can't just subtract from max value if negatives are included in the value range.

v=0.5
maxval=1
minval=-1

then if we just do:
1-0.5 => 0.5

This is not solarizing that value, which should actually become -0.5 if I understand the solarize operation correctl.

Would:

(maxval - minval) - (v - minval) - minval =>
1 - - 1 - 0.5 + 1 - -1 => 
2 - 0.5 + 2 => -0.5

work?

Reply:
> if you're in the range [-1, 1] a solarize operation is actually:
> 
> ```
> (2 - (v + 1)) - 1
> ```
> 
> Right? If I understand correctly you can't just subtract from max value if negatives are included in the value range.
> 
> v=0.5 maxval=1 minval=-1
> 
> then if we just do: 1-0.5 => 0.5
> 
> This is not solarizing that value, which should actually become -0.5 if I understand the solarize operation correctly.

It still works for negative ranges, like -1 and +1:
```python
min_value = -1
max_value = 1
# the equation is currently: max_value - x + min_value
# for value 0.5 the equation essentially becomes:
1 - 0.5 + (-1) => 1 - 0.5 - 1 => 0.5 - 1 = -0.5
```
Similary to other values, like 0.25 resulting in -0.25, (or reverse: -0.25 resulting in 0.25) this should be ok.
I added tests to test few example values.

Reply:
> The layer would then start by clipping to the value range, then apply `max_value - x` ?

@fchollet I am not sure about clipping input values. If it is decided that it's good to have I will add it, but personally I'm not sure if it's necessary. Please, let me know.
(This is probably what @sampathweb had in mind before, but I misunderstood him)

After that, yes, the value would be calculated by getting `max_value - x + min_value`.

Reply:
> @fchollet I am not sure about clipping input values. If it is decided that it's good to have I will add it, but personally I'm not sure if it's necessary. Please, let me know.

The problem with ignoring the value range is that output might silently be garbage if it's outside the expected range. There are possible alternatives to clipping:

- Scaling to the target range, by subtracting the min and dividing by the max. An issue here is that the scaling function may be different for different inputs, which messes with the input distribution.
- Erroring out if the input values fall outside the target range. This is the safest option.

> If this is the case, we can simply support a threshold parameter and use our Solarization layer for both

I think that's a great idea, it also avoids the logic redundancy with having two layers for similar use cases!

Reply:
We could put some reference in the docstring about  the inversion naming. E.g. `some paper etc..`

Reply:
Input values clipping has been added.

@LukeWood I removed the default constructor values and added optional threshold parameter. 

@bhack I mentioned AutoAugment in the docstring and which AutoAugment transform will be applied based on whether`threshold` parameter has been specified.